.github/workflows/docker.yaml

```diff
@@ -18,174 +19,106 @@ env:
-
-  robot:
-    strategy:
-      fail-fast: false
-      matrix:
-        ROS_DISTRO: [rolling]
-    runs-on: ubuntu-latest
-    permissions:
-      packages: write
-      contents: read
-    steps:
-      - name: Checkout repository
-        uses: actions/checkout@v4
+      -
+        # Add support for more platforms with QEMU (optional)
+        # https://github.com/docker/setup-qemu-action
+        name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
 
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v3.2.0
-
-      - name: Set up Docker Buildx
+      -
+        name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v3
 
-      - name: Log into registry
-        if: env.PUSH == 'true'
-        uses: docker/login-action@v3.3.0
-        with:
-          registry: ghcr.io
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
-
-      - name: Extract Docker metadata
-        if: env.PUSH == 'true'
-        id: meta
-        uses: docker/metadata-action@v5.5.1
-        with:
-          images: ghcr.io/${{ github.repository }}
-          tags: |
-            type=raw,value=${{ matrix.ROS_DISTRO }}-${{ github.job }}
-
-      - name: Build and push Docker image
-        uses: docker/build-push-action@v6.7.0
-        with:
-          context: .
-          file: .docker/Dockerfile
-          build-args: ROS_DISTRO=${{ matrix.ROS_DISTRO }}
-          target: ${{ github.job }}
-          tags: ${{ steps.meta.outputs.tags }}
-          labels: ${{ steps.meta.outputs.labels }}
-          push: ${{ env.PUSH }}
-          platforms: linux/amd64
-          #platforms: linux/amd64,linux/arm64
-
-  desktop:
-    strategy:
-      fail-fast: false
-      matrix:
-        ROS_DISTRO: [rolling]
-    runs-on: ubuntu-latest
-    permissions:
-      packages: write
-      contents: read
-    steps:
-      - name: Checkout repository
-        uses: actions/checkout@v4
-
-      - name: Log into registry
+      -
         if: env.PUSH == 'true'
+        name: Log into registry
         uses: docker/login-action@v3.3.0
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Extract Docker metadata
+      # buildx bake, unfortunately, requires lower-cased repository names
+      # This shell ... er, hack, creates a local variable containing
+      # a down-cased version of $GITHUB_REPOSITORY
+      #
+      - id: lower-repo
+        name: Repository to lowercase
+        run: |
+          echo "repository=${GITHUB_REPOSITORY@L}" >> $GITHUB_OUTPUT
+
+      # tbd:  not sure how to update this for buildx
+      #
+      # metadata-action can be used with bake:
+      #  https://github.com/docker/metadata-action?tab=readme-ov-file#bake-definition
+      #
+      # But not quite sure how to handle multiple targets given we don't iterate
+      # through the stages at this level (it happens within bake)
```

Comment:
The docs are a bit unclear to me, but it looks like we might need to configure the [`bake-file-tags`](https://github.com/docker/metadata-action?tab=readme-ov-file#outputs) to use the expected tags. Not quite sure though.

Reply:
After some consideration, it was more important to allow `buildx` to build the images effectively (and reduces caching confusion), so i unrolled the stages loop, now one buildx job builds all four stages.

This leads to four separate calls to metadata-action, one per image.   Verbose, but allows for adding distinct labels (semver, etc) for each image, if desired.

Reply:
I think that's a good call